Misc recipe additions/changes

Ayutac opened the pull request:
Added smooth basalt to basalts that can be centrifuged.
Added centrifuge recipes for rooted dirt and coarse dirt.
Added centrifuge recipe for moss block, which gives (uncompressed) plant ball.
Changed centrifuge recipe for grass block to give uncompr. plant ball instead of wheat seeds, eliminating the implication that every grass block holds wheat seeds.
Added grinder recipe to get small glowstone dust out of shroomlights (I forgot that in the fungus update)
Changed grinder recipe for gravel, now accepts all stone tool materials instead of only cobblestone.
Added a use for the andesite/diorite/granite dusts. Craft all 3 together with flint to get 4 gravel.
Added crafting 9 leaves/saplings of any kind into a plantball.
Added extractor recipe for green dye by using uncompr. plantball. Should never allow compressed plantball as that is out of juice, so to say.
Added extractor recipe for orange dye by using carrots.
Added extractor recipe for red dye by using beetroot.
Added extractor recipe for hanging roots by using rooted dirt.
Changed extractor recipe for stick to yield two sticks instead of one.
Changed electrolyzer recipe to accept anything tagged as sand instead of only the item sand, now also allowing red sand to be electrolyzed into silicon.

Commit:
Misc recipe additions/changes

Added smooth basalt to basalts that can be centrifuged.
Added centrifuge recipes for rooted dirt and coarse dirt.
Added centrifuge recipe for moss block, which gives (uncompressed) plant ball.
Changed centrifuge recipe for grass block to give uncompr. plant ball instead of wheat seeds, eliminating the implication that every grass block holds wheat seeds.
Added grinder recipe to get small glowstone dust out of shroomlights (I forgot that in the fungus update)
Changed grinder recipe for gravel, now accepts all stone tool materials instead of only cobblestone.
Added a use for the andesite/diorite/granite dusts. Craft all 3 together with flint to get 4 gravel.
Added crafting 9 leaves/saplings of any kind into a plantball.
Added extractor recipe for green dye by using uncompr. plantball. Should never allow compressed plantball as that is out of juice, so to say.
Added extractor recipe for orange dye by using carrots.
Added extractor recipe for red dye by using beetroot.
Added extractor recipe for hanging roots by using rooted dirt.
Changed extractor recipe for stick to yield two sticks instead of one.
Changed electrolyzer recipe to accept anything tagged as sand instead of only the item sand, now also allowing red sand to be electrolyzed into silicon.

modmuss50 reviewed (approved):
looks fine at a glace, will wait for someone else to take a look at the recipe changes balance wise and what not 👍

xanthian reviewed (commented):
the sapling > plantball should return more than leaves > plantball
otherwise looks good

Ayutac commented:
> the sapling > plantball should return more than leaves > plantball otherwise looks good

Good point. 2 or 3? I'm in favour of 3.

xanthian commented:
more - 10?
its only 10% chance to get sapling from leaves with fortune 3

drcrazy commented on src/main/resources/data/techreborn/recipes/centrifuge/coarse_dirt.json:
```diff
@@ -0,0 +1,25 @@
+      "count": 16
+    }
+  ],
+  "results": [
```
This could be 8xSand + 2xGravel + Flint to be in line with dirt

Ayutac replied:
didn't think of that one. my idea was that coarse dirt has a lot more gravel in it and there is no centrifuge recipe for gravel. Maybe 8 Gravel, 2 Sand, 1 Flint?
Anyway, I do not have strong feelings about this recipe, do as you like

drcrazy replied:
Yeah, 8xGravel, 2xSand and Flint looks good. Go for it.

Ayutac commented:
> more 

You say 10% chance for sapling per leave, I say 30 new leaves with one sapling. But if you insist... 5?

Commit:
Changed coarse dirt and sapling plantball thanks to feedback.

Commit:
Added centrifuge recipe for podzol

The idea here is that while there is a lot of plant stuff on the podzol stuff, it is not as fresh, hence only 1 plantball. On the other hand IRL podzol is rich of minerals and stuff, so two clay balls instead of one?

Ayutac commented:
Forgot the podzol lol. Also changed the other two debated recipes
